src/main/java/io/dropwizard/actors/connectivity/RMQConnection.java

```diff
@@ -33,15 +33,23 @@
     @Getter
     private Connection connection;
     private Channel channel;
+    private final MetricRegistry metricRegistry;
+    private final ExecutorService executorService;
 
-    public RMQConnection(RMQConfig config) {
+    public RMQConnection(RMQConfig config, MetricRegistry metricRegistry, ExecutorService executorService) {
         this.config = config;
+        this.metricRegistry = metricRegistry;
+        this.executorService = executorService;
     }
 
 
     @Override
     public void start() throws Exception {
         ConnectionFactory factory = new ConnectionFactory();
+        if (metricRegistry != null) {
+            factory.setMetricsCollector(new StandardMetricsCollector(metricRegistry));
```

Comment:
This will be a breaking change though. Only newer version of RMQ client supports metric collection.